src/accelerate/utils/__init__.py

```diff
@@ -192,24 +193,27 @@
     prepare_simple_launcher_cmd_env,
     prepare_tpu,
 )
-from .megatron_lm import (
-    AbstractTrainStep,
-    BertTrainStep,
-    GPTTrainStep,
-    MegatronEngine,
-    MegatronLMDummyDataLoader,
-    MegatronLMDummyScheduler,
-    MegatronLMOptimizerWrapper,
-    MegatronLMSchedulerWrapper,
-    T5TrainStep,
-    avg_losses_across_data_parallel_group,
-    gather_across_data_parallel_groups,
-)
-from .megatron_lm import initialize as megatron_lm_initialize
-from .megatron_lm import prepare_data_loader as megatron_lm_prepare_data_loader
-from .megatron_lm import prepare_model_optimizer_scheduler as megatron_lm_prepare_model_optimizer_scheduler
-from .megatron_lm import prepare_optimizer as megatron_lm_prepare_optimizer
-from .megatron_lm import prepare_scheduler as megatron_lm_prepare_scheduler
+
+
+if is_megatron_lm_available():
```

Comment:
For my understanding, this change is unrelated, right?

Reply:
Ah I missed that, had to scroll right to see the megatron part :D So it's not related to implementing the new workflow, but it is related to actually increasing the startup time.

Reply:
the check name is somewhat misleading, since a user may have megatron installed (i.e. available) but they aren't telling accelerate to use it.

perhaps `is_megatron_needed`?

for `available` - shouldn't it fail anyway long before if it's not available?

what I'm trying to communicate is for the sake of imports. `needed` and `available` are orthogonal to each other.

Reply:
No, it is always `available` and has been `available`, else we'd call it `needed` or `configured` as you say. 

`requires_` is for tests, we know we are using it so thus we use it.

`_available` simply means bring it into the environment if available.

Reply:
Its aggressive importing, yes. Which as noted in my last comment, will aim to fix when I can dedicate some time to refactoring our entire import schema in accelerate

Reply:
but the first thing the `is_megatron_available` check does is whether it's configured to be used, or am I misreading it?

https://github.com/huggingface/accelerate/blob/67b294398f4e3ab40a57a767f935a49a74e7f5d7/src/accelerate/utils/imports.py#L224

so it's not available - it's `is_wanted_and_available`